package/src/components/Channel/Channel.tsx

```diff
@@ -736,70 +736,61 @@ const ChannelWithContext = <
-    if (shouldSyncChannel) {
-      if (thread) {
-        const updatedThreadMessages =
-          (thread.id && channel && channel.state.threads[thread.id]) || threadMessages;
-        setThreadMessages(updatedThreadMessages);
-      }
-
-      if (channel && thread && event.message?.id === thread.id) {
-        const updatedThread = channel.state.formatMessage(event.message);
-        setThread(updatedThread);
-      }
-
-      if (event.type === 'typing.start' || event.type === 'typing.stop') {
-        copyTypingState();
-      } else if (event.type === 'message.read') {
-        copyReadState();
-      } else if (event.type === 'message.new') {
-        copyMessagesState();
-      } else if (channel) {
-        copyChannelState();
-      }
-    }
-  };
-
+  // subscribe to specific channel events
   useEffect(() => {
     const channelSubscriptions: Array<ReturnType<ChannelType['on']>> = [];
-    const clientSubscriptions: Array<ReturnType<StreamChat['on']>> = [];
+    if (channel && shouldSyncChannel) {
+      channelSubscriptions.push(channel.on('message.new', copyMessagesState));
+      channelSubscriptions.push(channel.on('message.read', copyReadState));
+      channelSubscriptions.push(channel.on('typing.start', copyTypingState));
+      channelSubscriptions.push(channel.on('typing.stop', copyTypingState));
+    }
+    return () => {
+      channelSubscriptions.forEach((s) => s.unsubscribe());
+    };
+  }, [channelId, shouldSyncChannel]);
 
-    if (!channel) return;
+  // subscribe to the generic all channel event
+  useEffect(() => {
+    const handleEvent: EventHandler<StreamChatGenerics> = (event) => {
+      if (shouldSyncChannel) {
+        if (thread) {
+          const updatedThreadMessages =
+            (thread.id && channel && channel.state.threads[thread.id]) || threadMessages;
+          setThreadMessages(updatedThreadMessages);
+        }
 
-    clientSubscriptions.push(
-      client.on('channel.deleted', (event) => {
-        if (event.cid === channel.cid) {
-          setDeleted(true);
+        if (channel && thread && event.message?.id === thread.id) {
+          const updatedThread = channel.state.formatMessage(event.message);
+          setThread(updatedThread);
         }
-      }),
-    );
 
-    if (enableOfflineSupport) {
-      clientSubscriptions.push(DBSyncManager.onSyncStatusChange(connectionChangedHandler));
-    } else {
-      clientSubscriptions.push(
-        client.on('connection.changed', (event) => {
-          if (event.online) {
-            connectionChangedHandler();
-          }
-        }),
-      );
-    }
+        // only update channel state if the events are not the previously subscribed useEffect's subscription events
+        if (
+          channel &&
+          event.type !== 'message.new' &&
+          event.type !== 'message.read' &&
+          event.type !== 'typing.start' &&
+          event.type !== 'typing.stop'
+        ) {
```

Comment:
```suggestion
        // only update channel state if the events are not the previously subscribed useEffect's subscription events
        const SKIP_UPDATE_EVENTS = [ 'message.new' , 'message.read', 'typing.start' , 'typing.stop']
        if (
          channel &&
          !SKIP_UPDATE_EVENTS.includes(event.type)
        ) {
```

Reply:
I cannot 100% agree. Creating an array and searching is expensive compared to simple comparisons :D Isnt it already readable?

Reply:
THis shouldn't make this much of a difference, it's not a must for me